components/supervisor/frontend/src/index.ts

```diff
@@ -10,9 +10,37 @@
  */
 import { IDEMetricsServiceClient, MetricsName } from "./ide/ide-metrics-service-client";
 IDEMetricsServiceClient.addCounter(MetricsName.SupervisorFrontendClientTotal).catch(() => {})
-window.addEventListener('error', () => {
-    IDEMetricsServiceClient.addCounter(MetricsName.SupervisorFrontendErrorTotal).catch(() => {})
-})
+
+declare global {
+    interface Window {
+        gitpodIDEErrors: Map<string, string>[];
+    }
+}
+
+window.addEventListener('error', (event) => {
+
+    const flags = new Map();
+
+    //@ts-ignore We take a look at what is the resource that was attempted to load;
+    const resourceSource = event?.target?.getAttribute('src') || event?.target?.getAttribute('href');
+
+    // If the event has a `target`, it means that it wasn't a script error
+    if (resourceSource && resourceSource.match(new RegExp(/\/build\/ide\/code:.{1,100}\/__files__\//g))) {
+        flags.set('resource', 'vscode-web-workbench');
+        flags.set('error', 'LoadError');
+    } else {
+        flags.set('error', 'Unknown');
```

Comment:
what if this is target but it does not match to vscode-web-workbench? Should we report it as `LoadError` or `Unknown`?

Reply:
I think it's best if we do LoadError with resource of unknown, so that we know a bit more about it